pkg/build/graph/analysis/bc.go

```diff
@@ -28,6 +32,12 @@ bc:
 		for _, istNode := range g.SuccessorNodesByEdgeKind(bcNode, buildedges.BuildOutputEdgeKind) {
 			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
 				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+				key := MissingRequiredRegistryWarning
+				reason := missingRegistryMessage
+				if !imageStreamNode.IsFound {
```

Comment:
> Missing image stream is another reason for a failed build push

I'm just looking at code flow and looks to me like I'll hit this block and then later I'm guaranteed to enter the `if` block lower down and I'm always going to return the same thing.  I don't see why it couldn't simply be

```go
if !imageStreamNode.IsFound {
    markers = append(markers, osgraph.Marker{
        RelatedNodes: []graph.Node{istNode},
        Severity: osgraph.WarningSeverity,
        Key: MissingImageStreamWarning,
        Message: fmt.Sprintf(..., MissingImageStreamMessage)
    })

    continue
}

if len(imageStreamNode.Status.DockerImageRepository) == 0 {
    // existing code here
}
```

As it is now, I have to follow the flow and have some semantic understanding to realize that one if block is forcing behavior down below.